tests/kernel/interrupt/src/static_isr.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <kernel_structs.h>
+#include "interrupt_util.h"
+
+#if defined(CONFIG_X86)
```

Comment:
The main problem with static ISR test is that we cannot "dynamically" (as in, at run-time) find an available IRQ, and the available IRQ numbers may vary across the SoCs for some archs, so we cannot hard-code them in the test (unless we `#ifdef` every available SoC or introduce some kind of automation scripts to handle this).

In that sense, I understand that this test needs to be restricted to certain supported arch/SoCs, but we should not be hard-coding x86 here in the static ISR test; instead, such constructs should be abstracted/defined in the `interrupt_util.h` (e.g. checking whether `TEST_IRQ_0_LINE` and `TEST_IRQ_1_LINE` are defined).

Reply:
I make some change, to put the trigger_irq() function in interrupt_util.h. If there are still things need to be improve, pls kindly let me know, thanks.